src/dcqc/target.py

```diff
@@ -104,9 +81,50 @@ def from_dict(cls, dictionary: SerializedObject) -> Target:
         Returns:
             The reconstructed target object.
         """
+        target_cls_name = dictionary["type"]
+        target_cls = BaseTarget.get_subclass_by_name(target_cls_name)
         dictionary = deepcopy(dictionary)
-        dictionary = cls.from_dict_prepare(dictionary)
+        dictionary = target_cls.from_dict_prepare(dictionary)
         files = [File.from_dict(d) for d in dictionary["files"]]
         id = dictionary["id"]
-        target = cls(*files, id=id)
+        target = target_cls(*files, id=id)
         return target
+
+    @classmethod
+    def get_base_class(cls):
+        """Retrieve base class."""
+        return BaseTarget
+
+
+@dataclass(init=False)
+class SingleTarget(BaseTarget):
+    """Single-file target."""
+
+    def __post_init__(self):
+        """Run validation checks after initialization."""
+        self.ensure_single_file()
+
+    def ensure_single_file(self):
```

Comment:
Another good catch. In fact, that's exactly the approach I tried at first. However, I was running into an impossible-to-debug issue in another class because of pydantic being used here, so I reverted the change. I can take another stab now that the rest of the PR is done.

Reply:
Interesting - no need to complicate things if this works.

Reply:
I tried it again, and I ran into the strange bug. We're going to skip Pydantic for now. I'll leave a comment.